torch/distributions/multivariate_normal.py

```diff
@@ -0,0 +1,85 @@
+import math
+from numbers import Number
+
+import torch
+from torch.autograd import Variable
+from torch.distributions.distribution import Distribution
+from torch.distributions.utils import broadcast_all
+
+class MultivariateNormal(Distribution):
+    r"""
+    Creates a multivariate normal (also called Gaussian) distribution
+    parameterized by a mean vector and a covariance matrix.
+    
+    The multivariate normal distribution can be parameterized either 
+    in terms of a positive definite covariance matrix :math:`\mathbf{\Sigma}` 
+    or a lower-triangular matrix :math:`\mathbf{L}` such that 
+    :math:`\mathbf{\Sigma} = \mathbf{L}\mathbf{L}^\top` as obtained via e.g. 
+    Cholesky decomposition of the covariance.
+
+    Example:
+
+        >>> m = MultivariateNormal(torch.zeros(2), torch.eye(2))
+        >>> m.sample()  # normally distributed with mean=`[0,0]` and cov=`I`
+        -0.2102
+        -0.5429
+        [torch.FloatTensor of size 2]
+
+    Args:
+        mean (Tensor or Variable): mean of the distribution
+        cov(Tensor or Variable): covariance matrix (sigma positive-definite).
```

Comment:
nit: It would be nice to stay maximally compatible with Tensorflow.distributions and name this `covariance_matrix`.

Reply:
Yeah, I just left this matching the scipy `mean` and `cov` because they were so much shorter. Happy to change to `loc` and `covariance_matrix` if that is what we've settled on.

Reply:
Yeah I'd like to keep the interfaces similar if possible, but I defer to your judgement here.